File: problems/3158_find_the_xor_of_numbers_which_appear_twice/solution.py
from typing import List


class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        from collections import Counter
        cnt = Counter(nums)
        res = 0
        for k, v in cnt.items():
            if v == 2:
                res ^= k
        return res

File: problems/3158_find_the_xor_of_numbers_which_appear_twice/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        unordered_map<int,int> cnt;
        for (int x : nums) cnt[x]++;
        int res = 0;
        for (auto& [k,v] : cnt) if (v == 2) res ^= k;
        return res;
    }
};
